src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -219,25 +293,48 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
     ReturnOnFailure(MdnsContexts::GetInstance().Has(sdCtx));
     LogOnFailure(__func__, err);
 
+    sdCtx->domainName = GetDomainFromHostName(hostname);
```

Comment:
So hold on.  Are there separate ResolveContext instances for the two domains?  Because if not, this is not making sense to me at all.

And looking above there are not.  So please remove the `domainName` member and fix all this code to work _without_ that.  I'll take another look once that's been done; stopping here.

Reply:
i didn't have the domain name in the context before but recent changes to the code made me add that specifically this part - https://github.com/project-chip/connectedhomeip/blob/master/src/platform/Darwin/DnssdContexts.cpp#L539

Reply:
There is no hostname here to get the domain from.

Reply:
@bzbarsky-apple Would it make sense to look for interfaces with domains we care about [here]( https://github.com/project-chip/connectedhomeip/blob/master/src/platform/Darwin/DnssdContexts.cpp#L539) that is call get interfaces with interfacekey - interface index + kLocalDot and  interface index + kOpenThreadDot